app/cafe.py

```diff
@@ -0,0 +1,24 @@
+import datetime
+
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError,
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if "vaccine" not in visitor:
+            raise NotVaccinatedError("All friends should be vaccinated")
+        if visitor["vaccine"]["expiration_date"] < datetime.date.today():
+            raise OutdatedVaccineError("Outdata vaccine")
+        if (
+            "wearing_a_mask" not in visitor
+            or visitor["wearing_a_mask"] is False
+        ):
+            raise NotWearingMaskError("All friends must be wear masks")
+        return f"Welcome to {self.name}"

```

Comment:
It's a code quality issue. In your function `visit_cafe`, you are checking whether certain keys are present in the `visitor` dictionary. However, if a key is not present, Python will raise a `KeyError` exception. To avoid this, you can use the `get` method of the dictionary, which returns `None` if the key is not present. Here is the algorithm for solving it: 1. Use the `get` method to get the value of the key. 2. Check if the value is `None` or not. If it is `None`, raise an exception. 3. If the value is not `None`, continue with the rest of the code.